pandas/io/formats/printing.py

```diff
@@ -199,11 +199,10 @@ def pprint_thing(
     -------
     str
     """
-
     def as_escaped_string(
         thing: Any, escape_chars: EscapeChars | None = escape_chars
     ) -> str:
-        translate = {"\t": r"\t", "\n": r"\n", "\r": r"\r"}
+        translate = {"\t": r"\t", "\n": r"\n", "\r": r"\r", "'": ""}
```

Comment:
I believe this will make the output here:

```python
df = pd.DataFrame({"'a": [1], "b": [2]}).set_index(["'a", "b"])
print(df.index.names)
```

be `['a', 'b']`. That is not correct.

Reply:
What would be the desired behavior?

Reply:
The index names must be accurate, the first being `'a`. I would think `['\'a', 'b']`.